osu.Game.Rulesets.Taiko/UI/TaikoMascotAnimation.cs

```diff
@@ -85,8 +85,12 @@ public ManualMascotTextureAnimation(TaikoMascotAnimationState state)
             }
 
             [BackgroundDependencyLoader]
-            private void load(ISkinSource skin)
+            private void load(ISkinSource source)
             {
+                ISkin skin = source.FindProvider(s => getAnimationFrame(s, state, 0) != null);
```

Comment:
Not sure these are right. I'd more be expecting to see the `FindProvider` call a level up (in `TaikoMascotAnimation` itself) and the resulting skin passed down to the animation states. Then again it raises the question of which particular asset should be queried to encompass all states.

Reply:
I dunno. This is how stable will work (since it just treats them the same way as animation lookups). I don't think there's generally a case this matters though - skins which provide part of the mascot generally provide all the states.